engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
+    implicit val actorSystem = CromwellServer.actorSystem
+    val materializedWfDescriptorFuture =
+      PromiseActor.askNoTimeout(MaterializeWorkflowDescriptorActor.MaterializeWorkflow(id, workflowSourceFiles), CromwellServer.workflowDescriptorMaterializeActor).mapTo[MaterializationResult]
+    materializedWfDescriptorFuture map {
+      case MaterializationSuccess(workflowDescriptor) => workflowDescriptor
+      case MaterializationFailure(error) => throw error
+    }
+  }
```

Comment:
I find that enhanced classes almost always make things more readable as saying something like `foo.someTransform.anotherTransform` fits the more modern functional dataflow-y style than `anotherTransform(someTransform(foo))